tests/lib/testers/rule-tester.js

```diff
@@ -621,9 +612,33 @@ describe("RuleTester", () => {
 
     });
 
+    it("throw an error when an unknown config option is included", () => {
+
+        assert.throws(() => {
+            ruleTester.run("no-eval", require("../../fixtures/testers/rule-tester/no-eval"), {
+                valid: [
+                    { code: "Eval(foo)", foo: "bar" }
+                ],
+                invalid: []
+            });
+        }, /ESLint configuration is invalid./);
```

Comment:
I think this would be tricky to do because `RuleTester` uses `linter.verify` to verify the config and run the rule at the same time, so it would need to be able to reliably distinguish between config errors and other types of thrown errors (e.g. rule crashes). Personally, I think "ESLint configuration is invalid" is reasonably clear (the issue in this case is that a test case contained an invalid key which did, in fact, get used as configuration), so IMO it might not be worth increasing complexity for a different message.

Also, it's worth noting that messages like this will still contain more information about what was wrong (in this case, it will complain about the "foo" key being invalid), so the user will still be given some direction for fixing the error.